### PR TITLE
Test pytorch 2.1.0 docker images on ci/cd

### PR DESCRIPTION
**Pull request by j316chuck**
# Description: 

Test pytorch 2.1.0 docker images on ci/cd for cpu and daily tests. Also fixes broken tests due to UserWarnings on torch 2.1.0 related to: 
- has_*** is deprecated. 
- enable safe_tensor is True. 

# Tests:
- CI/CD Tests for new nightly image all [pass](https://github.com/mosaicml/composer/actions/runs/5979443813/job/16223545853?pr=2469) ✅

### CODE REVIEW

File: .github/workflows/pr-gpu.yaml [hunk 1]
```diff
@@ -26,6 +26,11 @@ jobs:
             markers: 'not daily and not remote and gpu and (vision or not vision) and (doctest or not doctest)'
             pytest_command: 'coverage run -m pytest'
             composer_package_name: 'mosaicml'
+          - name: 'gpu-3.10-2.1-pytorch-nightly'
```
Comment (mvpatel2000) [inline .github/workflows/pr-gpu.yaml:29]: Let's skip this for now... we don't really have GPUs lying around

File: .github/workflows/pr-cpu.yaml [hunk 2]
```diff
@@ -31,6 +31,11 @@ jobs:
             markers: 'not daily and not remote and not gpu and not vision and not doctest'
             pytest_command: 'coverage run -m pytest'
             composer_package_name: 'mosaicml'
+          - name: 'cpu-3.10-2.1'
```
Comment (mvpatel2000) [inline .github/workflows/pr-cpu.yaml:34]: can we copy this into daily as well?

Comment (j316chuck) [reply .github/workflows/pr-cpu.yaml:34]: added

File: .github/workflows/pr-gpu.yaml (hunk 1 above)
Comment (j316chuck) [reply .github/workflows/pr-gpu.yaml]: removed

Comment (j316chuck) [reply .github/workflows/pr-gpu.yaml]: removed

File: .github/workflows/daily.yaml [hunk 3]
```diff
@@ -53,6 +53,11 @@ jobs:
             markers: 'daily and (remote or not remote) and not gpu and not vision and not doctest'
             pytest_command: 'coverage run -m pytest'
             composer_package_name: 'composer'
+          - name: 'daily-cpu-3.10-2.1'
```
Comment (mvpatel2000) [inline .github/workflows/daily.yaml:56]: this file also needs the normal case (in pr-cpu) at the top

Comment (j316chuck) [reply .github/workflows/daily.yaml:56]: good catch, added 👍

**Review by mvpatel2000 — APPROVED** [review]
LGTM! Thanks

**Pull request merged**